### PR TITLE
Release

### LINKED ISSUE #44: Handeling protocol fees
The bot currently assumes all trading fees (usually .3% on dexs) remain in the pools for the LP rewards. However, most protocols use something called "protocol fee"  or "burn fee", which is a portion of the trading fees. This portion leaves the pool and therefore influences the expected state. This should be handled properly

### LINKED ISSUE #48: tendermint-rpc Bad status on response: 500
If you want to report a security issue, please follow our security policy: `https://github.com/White-Whale-Defi-Platform/migaloo-core/blob/main/SECURITY.md`

---
## Steps to Reproduce

1. See Log 
2. `npm start` and wait around 350-600 Blocks 
3. tendermint-rpc Bad status response: 500 

**Expected results:** 
App waits x seconds and tries again.

**Actual results:** 
App is crashing. (Log)

---

### Additional context
Add any other context about the problem here, code snippets, json responses and so on.

</details>

<details>
  <summary>Logs</summary>

```
/home/dietpi/white-whale-bots/node_modules/@cosmjs/tendermint-rpc/build/rpcclients/http.js:10
throw new Error(`Bad status on response: ${res.status}`);   
^

Error: Bad status on response: 500
at filterBadStatus (/home/dietpi/white-whale-bots/node_modules/@cosmjs/tendermint-rpc/build/rpcclients/http.js:10:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Node.js v19.6.0
```

### LINKED ISSUE #51: Handle send message in mempool properly
When the bot tries to apply mempool transactions on local state, it tries to decode 'MsgSend' objects among other. 
Usually, these send messages happen on a cw20 token and result in a swap of the token, but apparently sometimes they also result in a 'withdraw_liquidity' message. When this occurs, the bot tries to swap an LP token with the assets of that specific pool and will result in an error: 
```
TypeError: Cannot read properties of undefined (reading 'amount')
    at applyTradeOnPool (/usr/src/app/out/core/types/base/pool.js:51:28)
    at applyMempoolTradesOnPools (/usr/src/app/out/core/types/base/pool.js:177:33)
    at SkipLoop.step (/usr/src/app/out/core/types/arbitrageloops/skipLoop.js:62:58)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async main (/usr/src/app/out/index.js:118:9)
```

This can be fixed to tighten up the check isSendMsg() in mempool.ts

### PR DESCRIPTION
**Pull request by SirTLB**
Releasing developments into main. see commit log for info. 
fixes #61, fixes #51, fixes #48, fixes #44 
Other features:
-- discord logger
-- updated example files
-- docker compose
-- sign of life to environment variables

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,29 +4,32 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "prebuild": "npm run clean && echo \"building...\"",
```
Comment (carlthelooper) [inline package.json:7]: Do we want to add linting to help with future contributions?

```
"prebuild": "npm run clean && npm run lint && echo \"building...\"",

"lint": "eslint . --ext .ts",
"lint:fix": "eslint . --ext .ts --fix",
```

File: package.json [hunk 2]
```diff
@@ -47,6 +50,7 @@
     "@cosmjs/tendermint-rpc": "^0.29.4",
     "@skip-mev/skipjs": "^1.1.0",
```
Comment (carlthelooper) [inline package.json:51]: Any interest in adding #54 to update skip and cosm?  Or wait until a future release?

Comment (SirTLB) [reply package.json:51]: I have a question out with skip how to solve this signer issue. I'd really rather not use ts-ignore to get a private signer out.

Comment (carlthelooper) [reply package.json:51]: Oh good, I would prefer they fixed their issue as well.  Thanks for the update!

File: package.json (hunk 1 above)
Comment (SirTLB) [reply package.json:7]: how would that work, would someone need to run it manually?

**Review by nick134-bit — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (SirTLB) [reply package.json:7]: Alright seems good to me. I can try to add that and ask for a review again.

File: .env.juno.example [hunk 3]
```diff
@@ -1,89 +1,90 @@
...
 BASE_DENOM="ujuno" # The asset denom to be used as base asset. This should be the denom of a Native Token only.
 CHAIN_PREFIX="juno"
-RPC_URL="https://juno-rpc.reece.sh"
+RPC_URL="https://juno-rpc.reece.sh" #change this
 GAS_UNIT_PRICE="0.0025"
 FLASHLOAN_ROUTER_ADDRESS="juno1qa7vdlm6zgq3radal5sltyl4t4qd32feug9qs50kcxda46q230pqzny48s"
-
 FACTORIES_TO_ROUTERS_MAPPING ='{"factory":"juno14m9rd2trjytvxvu4ldmqvru50ffxsafs8kequmfky7jh97uyqrxqs5xrnx","router":"juno128lewlw6kv223uw4yzdffl8rnh3k9qs8vrf6kef28579w8ygccyq7m90n2"},
 {"factory":"juno16adshp473hd9sruwztdqrtsfckgtd69glqm6sqk0hc4q40c296qsxl3u3s","router":"juno1pctfpv9k03v0ff538pz8kkw5ujlptntzkwjg6c0lrtqv87s9k28qdtl50w"}'
 FLASHLOAN_FEE="0.3" #in %
```
Comment (carlthelooper) [inline .env.juno.example:31]: Do we need to set this to .15 now?

File: package.json [hunk 4]
```diff
@@ -4,29 +4,34 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "prebuild": "npm run clean && echo \"building...\"",
```
Comment (carlthelooper) [inline package.json:7]: Add the lint step?

`"prebuild": "npm run clean && npm run lint && echo \"building...\"",`

Comment (SirTLB) [reply package.json:7]: yes that was my question, wasnt sure if we'd want that. for me i get a lot of lf <> crlf issues... maybe we should handle those at repo level not sure

File: .env.juno.example (hunk 3 above)
Comment (SirTLB) [reply .env.juno.example:31]: yes we can change that

File: package.json (hunk 4 above)
Comment (carlthelooper) [reply package.json:7]: Oh sorry, I misunderstood and thought you just missed it.  I'm open either way, just think it is nice to have in general. So having it as part of the build step or incorporating it into a merge check in GH works for me.

**Review by carlthelooper — DISMISSED** [review]

File: Dockerfile [hunk 5]
```diff
@@ -1,17 +1,14 @@
...
 WORKDIR /usr/src/app
 
 COPY package*.json ./
-
 RUN npm install
 
-COPY . .
-
+COPY tsconfig.json ./
+COPY src ./src
```
Comment (SirTLB) [inline Dockerfile:11]: @carlthelooper I think we also need to COPY .env ./ right?

**Review by zJuuu — DISMISSED** [review]

File: Dockerfile (hunk 5 above)
Comment (SirTLB) [reply Dockerfile:11]: tho you can also use this dockerfile wihtout compose... just use docker build . 
i'd say we make sure this file works anyways

Comment (carlthelooper) [reply Dockerfile:11]: To stay consistent for spinning up a container with docker-compose or docker you'd need to leave copying the .env out of the dockerfile.  For spinning up the container with docker you'd just add a param for including the `env_file` or including all the env variables if you really wanted to.

File: package.json [hunk 6]
```diff
@@ -4,29 +4,34 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "prebuild": "prebuild": "npm run clean && npm run lint && echo \"building...\"",
```
Comment (carlthelooper) [inline package.json:7]: extra `"prebuild":` 😄

Comment (carlthelooper) [reply package.json:7]: Let's also change to `lint:fix` which clears up basic formatting issues that are causing trouble when creating a docker image on certain systems.

File: Dockerfile [hunk 7]
```diff
@@ -1,17 +1,14 @@
...
+RUN mkdir /usr/src/app
 WORKDIR /usr/src/app
 
 COPY package*.json ./
-
 RUN npm install
 
-COPY . .
-
+COPY tsconfig.json ./
```
Comment (carlthelooper) [inline Dockerfile:10]: Need to add another line `COPY .eslintrc ./` since we added linting to the prebuild.

**Review by SirTLB — COMMENTED** [review]
changed dockerfile and package

**Review by carlthelooper — APPROVED** [review]
🚢

File: src/core/logging/logger.ts [hunk 8]
```diff
@@ -10,22 +11,35 @@ export class Logger {
 	private botConfig: BotConfig;
 	public discordLogger?: DiscordLogger;
 	public slackLogger?: SlackLogger;
+	private telegramLogger?: TelegramLogger;
 
 	// Codes that are not sent to external sources (discord, slack)
-	private externalExemptCodes: Array<number> = [4, 5, 6, 8];
+	private externalExemptCodes: Array<number> = [];
```
Comment (Sen-Com) [inline src/core/logging/logger.ts:17]: Why have this line if it's empty by default and there's no env var to change it?

**Review by Sen-Com — APPROVED** [review]

**Pull request merged**